### PR TITLE
Run3-sim62 Add hits for the new Totem T2 detector

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a new hit collection for the newly proposed Totem T2 detector for the Run3 application

#### PR validation:

Tested with runTheMatrix.py and the new hit collection is tested with #29268 and a cfg file in SimG4CMS/Forward

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Forward/src/TotemT2ScintSD.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#include "G4VProcess.hh"
+#include "G4ios.hh"
+#include "G4Cerenkov.hh"
+#include "G4ParticleTable.hh"
+#include "CLHEP/Units/GlobalSystemOfUnits.h"
+#include "CLHEP/Units/GlobalPhysicalConstants.h"
+#include "Randomize.hh"
+#include "G4Poisson.hh"
+
+#define EDM_ML_DEBUG
```
Comment (cvuosalo) [inline SimG4CMS/Forward/src/TotemT2ScintSD.cc]:
```suggestion
// #define EDM_ML_DEBUG
```

**Comment by cvuosalo** [comment]
@bsunanda You might want to put "Totem" in the PR title to inform the uninitiated.

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
@bsunanda , for  me it is strange, that Geometry/ForwardSimData/data/totemsensT2/2021/v2/totemsensT2.xml is inside "2021/v2". Why not directly inside "2021"? An agreement as I understand it is that inside 2021 we have the baseline for the Run-3, when the geometry will be frozen each new update is inside "v1", "v2"... 

Second moment: it is not possible to overwrite cuts for the world volume here. If for a sub-detector we need a special set of cuts we create a new region. I would propose making Totem2 region for this purpose.

**Comment by bsunanda** [comment]
There was a file called Geometry/ForwardSimData/data/totemsensT2/2021/v1/totemsensT2.xml - which was indeed a file for Run2 and I moved it to Geometry/ForwardSimData/data/totemsensT2/2018 now. This detector was used in Run2 and is decommissioned for Run3. I have defined a new cut region "TotemT2Region" for which cuts are defined

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Forward/src/TotemT2ScintSD.cc (hunk 1 above)
Comment (bsunanda) [reply SimG4CMS/Forward/src/TotemT2ScintSD.cc]: Done

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**